base/show.jl

```diff
@@ -19,24 +19,28 @@ immutable IOContext{IO_t <: IO} <: AbstractPipe
     end
 end
 
-"""
-    IOContext(io::IO; properties...)
-
-The same as `IOContext(io::IO, KV::Pair)`, but accepting properties as keyword arguments.
-"""
-IOContext(io::IO; kws...) = IOContext(IOContext(io, ImmutableDict{Symbol,Any}()); kws...)
-function IOContext(io::IOContext; kws...)
-    for (k, v) in kws
-        io = IOContext(io, k, v)
-    end
-    io
-end
+# TODO: This defeats the minimal-allocation properties of the no-op constructor
+#"""
+#    IOContext(io::IO; properties...)
+#
+#The same as `IOContext(io::IO, KV::Pair)`, but accepting properties as keyword arguments.
+#"""
+#IOContext(io::IO; kws...) = IOContext(IOContext(io); kws...)
+#function IOContext(io::IOContext; kws...)
```

Comment:
Not easily. The problem is that it replaces the primitive `IOContext(io)` constructor (which would otherwise be inlined), with several kwarg calls. (since the kwarg method needs to be the same as the no-keyword args function)